pandas/core/indexes/period.py

```diff
@@ -205,6 +210,18 @@ def _resolution_obj(self) -> Resolution:
         **_shared_doc_kwargs,
     )
     def asfreq(self, freq=None, how: str = "E") -> Self:
+        if isinstance(freq, str):
+            offset = to_offset(freq, is_period=True)
+            if hasattr(offset, "_period_dtype_code"):
+                freq = freq_to_period_freqstr(offset.n, offset.name)
+            elif offset.name == freq.replace(f"{offset.n}", ""):
+                raise ValueError(
+                    f"Invalid offset: '{offset.name}' for converting time series "
+                    f"with PeriodIndex."
+                )
+            else:
+                raise ValueError(INVALID_FREQ_ERR_MSG.format(f"{freq}"))
+
         arr = self._data.asfreq(freq, how)
```

Comment:
can you follow this `asfreq` and put this validation even deeper?

Reply:
thanks, I did as you suggested and moved this validation to `to_offset`. It works very well.